surveys/views.py

```diff
@@ -316,8 +319,31 @@ def form_valid(self, form):
                                'collect_data'
                                )
 
+        # add survey questions from template
+        template = form.cleaned_data['survey_template']
+
+        if template:
+            template_questions = fobi_models.FormElementEntry.objects.filter(form_entry=template)
```

Comment:
How much of an extra lift would it be to add some validation logic to ensure that only intercept surveys use intercept templates, and vice versa? I'm thinking of potentially two changes:

1. Some conditional logic in the survey creation form that shows/hides options depending on whether the form type was selected as `intercept` or `observational`
2. Backend validation on the `SurveyCreateForm.clean_survey_template()` method to make sure the `type` of the survey matches the `type` of the template (as a safeguard in case the frontend conditional formatting fails, e.g. if the user selects the template before they select the survey type)

Reply:
Added backend validation! I had thought about this before and added the type of the template in parenthesis to tip users off. Conditional logic (#1) is a heavier lift because it will involve writing JavaScript in the template. If it's ok with you, I don't think that's a high need.